From a0ff39aef11e6e390f01b395c0f7063f6179b98e Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Tue, 7 Aug 2012 11:36:47 +0200
Subject: [PATCH] fdo#51324 lp#1017125 rhbz#806236 rhbz#823272: erase on
 invalid iterators

* the iterators returned by equal_range are invalidated by modification
  (which deleting them clearly is)
* possibly only a partial fix as there is more abuse of equal_range in that
  file

Change-Id: Ie9ba01e91cb449a32d77a69590bda2125f0924e1
---
 sd/source/ui/toolpanel/TaskPaneFocusManager.cxx |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sd/source/ui/toolpanel/TaskPaneFocusManager.cxx b/sd/source/ui/toolpanel/TaskPaneFocusManager.cxx
index 6be7076..4f9698a 100644
--- a/sd/source/ui/toolpanel/TaskPaneFocusManager.cxx
+++ b/sd/source/ui/toolpanel/TaskPaneFocusManager.cxx
@@ -226,8 +226,7 @@ void FocusManager::RemoveLinks (::Window* pWindow)
     pWindow->RemoveEventListener (LINK (this, FocusManager, WindowEventListener));
 
     // Remove the links from the given window.
-    ::std::pair<LinkMap::iterator,LinkMap::iterator> aCandidates(mpLinks->equal_range(pWindow));
-    mpLinks->erase(aCandidates.first, aCandidates.second);
+    mpLinks->erase(pWindow);
 
     // Remove links to the given window.
     bool bLinkRemoved;
-- 
1.7.9.5

